eng/build.yml

```diff
@@ -141,14 +152,21 @@ jobs:
           env:
             Token: $(dn-bot-dnceng-artifact-feeds-rw)
 
+    - ${{ if and(eq(parameters.architecture, 'arm64'), eq(parameters.osGroup, 'MacOS')) }}:
+      - script: /bin/bash -c "sudo xcode-select -s /Applications/Xcode_12.4.app/Contents/Developer"
```

Comment:
Is there a reason why we're specifically using Xcode 12.4 here?

Reply:
In diagnostics it's mostly needed for LLDB testing.